packages/ui-transfer-demo2/lib/demo.machine.ts

```diff
@@ -168,6 +180,21 @@ export const demoMachine: MachineConfig<DemoContext, DemoStateSchema, DemoEvent>
       on: {
         [DemoEvents.AUTHENTICATED]: {
           actions: assign({ session: (context, event) => event.session }),
+          target: DemoStates.LOADING_TOKEN,
+        },
+      },
+    },
+
+    [DemoStates.LOADING_TOKEN]: {
+      invoke: {
+        src: loadToken,
+      },
+      on: {
+        [DemoEvents.TOKEN_LOADED]: {
+          actions: assign((context, event) => ({
+            ... context,
+            token: event.token,
+          })),
```

Comment:
This should also work...
```suggestion
          actions: assign({ token: (context, event) => event.token }),
```